app/models/edition.rb

```diff
@@ -604,19 +601,17 @@ def body_required?
   end
 
   def has_associated_needs?
-    associated_needs.any?
+    respond_to?(:need_ids) && need_ids.try(:any?)
   end
 
   def associated_needs
```

Comment:
It seems like this should either all go in `DetailedGuide` if that is the only format that has needs or else it should all go in the base class (or preferably a separate class as `Edition` is already huge) if we're going to allow all formats to have needs now they're in the publishing API. Also, all `Edition`s will respond to `need_ids` and return an `Array` so if we are going to have `has_associated_needs?` it's unnecessarily defensive.